File: java/senscord/src/senscord_properties/CurrentFrameNumProperty.java
/*
 * SPDX-FileCopyrightText: 2019 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package senscord_properties;

import com.sun.jna.Structure;
import java.util.Arrays;
import java.util.List;

/**
 * Property for the current buffering frames.
 * 
 * @see "senscord::CurrentFrameNumProperty"
 */
public class CurrentFrameNumProperty extends Structure {
    public int arrived_number;
    public int received_number;

    public static class ByReference extends CurrentFrameNumProperty
            implements Structure.ByReference {
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("arrived_number", "received_number");
    }

    /**
     * Get arrived frame number.
     * 
     * @return arrived number
     */
    public int GetArrivedNumber() {
        return arrived_number;
    }


    /**
     * Get received frame number.
     * 
     * @return received frame number
     */
    public int GetReceivedNumber() {
        return received_number;
    }

}

File: csharp/lib/Property/CurrentFrameNumProperty.cs
/*
 * SPDX-FileCopyrightText: 2017-2022 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

using MessagePack;

namespace SensCord
{
    /// <summary>
    /// Property for the current buffering frames.
    /// </summary>
    [MessagePackObject]
    public class CurrentFrameNumProperty : IBaseProperty
    {
        /// <summary>
        /// Property key.
        /// </summary>
        public static readonly string ConstKey = "current_frame_num_property";

        /// <summary>
        /// Property key.
        /// </summary>
        [IgnoreMember]
        public string Key { get; } = ConstKey;

        /// <summary>
        /// Arrived number.
        /// </summary>
        [Key("arrived_number")]
        public int ArrivedNumber { get; set; }

        /// <summary>
        /// Received number.
        /// </summary>
        [Key("received_number")]
        public int ReceivedNumber { get; set; }

        /// <summary>
        /// To string.
        /// </summary>
        /// <returns>A string that represents the current object.</returns>
        public override string ToString()
        {
            return $"{this.Key}(arrived={this.ArrivedNumber}, received={this.ReceivedNumber})";
        }
    }
}
